File: Projekti/SQLAlchemy-Benchmark/repository/profile_repository.py
from model.profile import Profile
from util.stopwatch import Stopwatch


class ProfileRepository:

    def __init__(self, session):
        self.session = session

    def delete_profiles_by_phone_number(self, number):
        profiles_to_delete = self.session.query(Profile).filter(
            Profile.phone_number.like(number + '%')).all()
        for profile in profiles_to_delete:
            self.session.delete(profile)
        self.session.rollback()

File: Projekti/Hibernate-Benchmark/src/main/java/com/example/HibernateBenchmark/repository/ProfileRepository.java
package com.example.HibernateBenchmark.repository;

import com.example.HibernateBenchmark.model.Profile;
import com.example.HibernateBenchmark.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

public interface ProfileRepository extends JpaRepository<Profile, Integer> {

    @Modifying
    @Transactional
    @Query("DELETE FROM Profile p WHERE p.phoneNumber LIKE :prefix%")
    int deleteProfilesByPhonePrefix(@Param("prefix") String prefix);
}
